schemas/rust-toolchain.toml.json

```diff
@@ -0,0 +1,364 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "title": "Schema for rust-toolchain.toml",
+  "description": "https://rust-lang.github.io/rustup/overrides.html#the-toolchain-file",
+  "x-taplo-info": {
+    "authors": [
+      "ian-h-chamberlain (https://github.com/ian-h-chamberlain)"
+    ],
+    "patterns": [
+      "^(.*(/|\\\\)rust-toolchain([.]toml)?)$"
+    ]
+  },
+  "properties": {
+    "toolchain": {
+      "description": "The Rust toolchain to be used.",
+      "type": "object",
+      "minProperties": 1,
+      "oneOf": [
+        {
+          "properties": {
+            "channel": {
+              "description": "The release channel of Rust to use.",
+              "type": "string",
+              "oneOf": [
+                {
+                  "enum": [
+                    "stable",
+                    "beta",
+                    "nightly"
+                  ]
+                },
+                {
+                  "$comment": "Empty schema allows for custom rustup toolchains, specific nightlies, etc."
+                }
+              ],
+              "x-taplo": {
+                "links": {
+                  "key": "https://rust-lang.github.io/rustup/concepts/channels.html"
+                },
+                "docs": {
+                  "main": "Rust is released to three different \"channels\": stable, beta,\nand nightly."
```

Comment:
There were a couple instances where I wasn't sure if I should put the upstream docs into `x-taplo.docs.main` or just `description`, is there a preference on this?

Reply:
I'd say avoid taplo-specific extensions whenever possible, so prefer `description` in this case. The main reason why `x-taplo.docs.main` exists is that `x-taplo` is carried over `$ref`s in the current implementation, so it allows overriding the docs for referenced schemas, but I'm revisiting this decision in the next version as it complicates things.